### DISCUSSION TITLE
Why we are building 120-500M models

### DISCUSSION THREAD

**Discussion by techaarvam** [techaarvam/byom_workshop#6] [General]
In the upcoming workshop (details in the comments), why are we building 120-500M models?

Dreaded OOMs, very slow training -> there are techniques to stay away from both of these parts of town.

Mandatory techniques:

1. torch.compile

2. AMP - Automatic Mixed Precision

3. Adaptive-softmax-entropy instead of full-cross-entropy or chunked-cross-entropy.

4. Fused QKV and SDPA (scaled-dot-product-attention) or Flash-Attention

Probably required techniques:

1. gradient accumulation

2. gradient checkpointing.

Other things to not forget:

1. gradient clipping.

2. checkpoint pth - save/restore the model on failure and continue training.

3. logging verbosity and essential instrumentation.

Did I miss anything, add in the comments.
